apollo-federation/src/operation/mod.rs

```diff
@@ -1082,8 +1172,6 @@ mod field_selection {
     pub(crate) struct Field {
         data: FieldData,
         key: SelectionKey,
-        #[serde(serialize_with = "crate::display_helpers::serialize_as_debug_string")]
-        sorted_arguments: Arc<Vec<Node<executable::Argument>>>,
```

Comment:
This is a key motivator for the change: not requiring this additional field in the `Field` struct means we can generalise/simplify selection key structures in the future.